src/test/java/com/digirati/taxonomy/manager/lookup/persistence/SkosTranslatorTest.java

```diff
@@ -0,0 +1,203 @@
+        String expected = IOUtils.toString(expectedStream, StandardCharsets.UTF_8).trim();
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void shouldExtractConceptsFromRdf() {
+        // Given
+        InputStream testConceptsInput = load("test-concepts-input.jsonld");
+
+        // When
+        RdfModel actual = underTest.translate(testConceptsInput, null, SkosFileType.JSON_LD);
+
+        // Then
+        assertEquals(conceptsRdfModel, actual);
+    }
+
+    @Test
+    void shouldSerialiseConceptsToRdf() throws IOException {
+        // When
+        Model actualModel = underTest.translate(conceptsRdfModel);
+        StringWriter actualWriter = new StringWriter();
+        actualModel.write(actualWriter, SkosFileType.JSON_LD.getFileTypeName());
+        String actual = actualWriter.toString().trim();
+
+        // Then
+        InputStream expectedStream = load("test-concepts-output.jsonld");
+        String expected = IOUtils.toString(expectedStream, StandardCharsets.UTF_8).trim();
+
+        assertEquals(expected, actual);
+    }
+
+    private static RdfModel createConceptSchemesModel() {
+        ConceptSchemeModel conceptSchemeModel =
+                new ConceptSchemeModel().setIri("http://example.com/");
+
+        ConceptModel one =
+                new ConceptModel()
+                        .setIri("http://example.com/concept#1")
+                        .setPreferredLabel(
+                                "[{\"language\":\"fr\",\"value\":\"un\"},{\"language\":\"en\",\"value\":\"one\"}]");
+        ConceptModel two =
+                new ConceptModel()
+                        .setIri("http://example.com/concept#2")
+                        .setPreferredLabel(
+                                "[{\"language\":\"fr\",\"value\":\"deux\"},{\"language\":\"en\",\"value\":\"two\"}]");
+
+        ConceptSemanticRelationModel schemeHasTopConceptOne =
+                new ConceptSemanticRelationModel()
+                        .setSourceId("http://example.com/")
+                        .setTargetId("http://example.com/concept#1")
+                        .setRelation(SemanticRelationType.HAS_TOP_CONCEPT)
+                        .setTransitive(false);
+        ConceptSemanticRelationModel schemeHasTopConceptTwo =
+                new ConceptSemanticRelationModel()
+                        .setSourceId("http://example.com/")
+                        .setTargetId("http://example.com/concept#2")
+                        .setRelation(SemanticRelationType.HAS_TOP_CONCEPT)
+                        .setTransitive(false);
+
+        return new RdfModel(
+                Arrays.asList(one, two),
+                Collections.singletonList(conceptSchemeModel),
+                Arrays.asList(schemeHasTopConceptTwo, schemeHasTopConceptOne));
+    }
+
+    private static RdfModel createConceptsModel() {
+        ConceptModel one =
+                new ConceptModel()
+                        .setIri("http://example.com/concept#1")
```

Comment:
Same again with the fluent-builder pattern. A huge constructor is preferable over this.

Reply:
To clarify: the semantics of a huge constructor are better. Syntactically, a builder might look better, but a builder and huge constructor aren't mutually exclusive.

```java
ConceptModel.Builder builder = ConceptModel.builder();
builder.set(...)
ConceptModel concept = builder.build(); // throws exception on malformed concept
```